packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -192,64 +204,142 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
+      assert(child.parent == null);
+    }
+  }
+
+  @override
+  void attach(PipelineOwner owner) {
+    super.attach(owner);
+    for (RenderBox child in _keepAliveBucket.values)
+      child.attach(owner);
+  }
+
+  @override
+  void detach() {
+    super.detach();
+    for (RenderBox child in _keepAliveBucket.values)
+      child.detach();
+  }
+
+  @override
+  void redepthChildren() {
+    super.redepthChildren();
+    for (RenderBox child in _keepAliveBucket.values)
+      redepthChild(child);
+  }
+
+  @override
+  void visitChildren(RenderObjectVisitor visitor) {
+    super.visitChildren(visitor);
+    for (RenderBox child in _keepAliveBucket.values)
+      visitor(child);
+  }
+
   /// Called during layout to create and add the child with the given index and
   /// scroll offset.
   ///
   /// Calls [RenderSliverBoxChildManager.createChild] to actually create and add
-  /// the child.
+  /// the child if necessary. The child may instead be obtained from a cache;
+  /// see [SliverMultiBoxAdaptorParentData.keepAlive].
   ///
-  /// Returns false if createChild did not add any child, otherwise returns
-  /// true.
+  /// Returns false if there was no cached child and `createChild` did not add
+  /// any child, otherwise returns true.
   ///
   /// Does not layout the new child.
   ///
-  /// When this is called, there are no children, so no children can be removed
-  /// during the call to createChild. No child should be added during that call
-  /// either, except for the one that is created and returned by createChild.
+  /// When this is called, there are no visible children, so no children can be
+  /// removed during the call to `createChild`. No child should be added during
+  /// that call either, except for the one that is created and returned by
+  /// `createChild`.
   @protected
   bool addInitialChild({ int index: 0, double layoutOffset: 0.0 }) {
     assert(_debugAssertChildListLocked());
     assert(firstChild == null);
     bool result;
-    invokeLayoutCallback<SliverConstraints>((SliverConstraints constraints) {
-      assert(constraints == this.constraints);
-      _childManager.createChild(index, after: null);
-      if (firstChild != null) {
-        assert(firstChild == lastChild);
-        assert(indexOf(firstChild) == index);
-        final SliverMultiBoxAdaptorParentData firstChildParentData = firstChild.parentData;
-        firstChildParentData.layoutOffset = layoutOffset;
-        result = true;
-      } else {
-        childManager.setDidUnderflow(true);
-        result = false;
-      }
-    });
+    _createOrObtainChild(index, after: null);
+    if (firstChild != null) {
+      assert(firstChild == lastChild);
+      assert(indexOf(firstChild) == index);
+      final SliverMultiBoxAdaptorParentData firstChildParentData = firstChild.parentData;
+      firstChildParentData.layoutOffset = layoutOffset;
+      result = true;
+    } else {
+      childManager.setDidUnderflow(true);
+      result = false;
+    }
```

Comment:
Since this is no longer a closure, you could kill off `bool result` and return the literals instead. 

I suspect I've got some sort of obsession relating to shortening lines code though -- probably dates back to those one-liners in the back of _Compute!_ magazine in the 80s.